core/src/main/scala/dagr/core/execsystem/TaskTracker.scala

```diff
@@ -0,0 +1,344 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package dagr.core.execsystem
+
+import java.nio.file.Path
+import java.sql.Timestamp
+
+import dagr.DagrDef._
+import dagr.core.execsystem.TaskStatus._
+import dagr.core.execsystem.TaskManagerLike.BaseInfoAndNode
+import dagr.core.execsystem.TaskTracker.InfoAndNode
+import dagr.core.tasksystem.Task
+import dagr.core.util.{BiMap, LazyLogging}
+
+import scala.collection.mutable
+
+object TaskTracker {
+  case class InfoAndNode(info: TaskExecutionInfo, node: GraphNode) extends BaseInfoAndNode(info=info, node=node)
+}
+
+/** Tracks tasks during their execution */
+trait TaskTracker extends TaskManagerLike with LazyLogging {
+
+  /** Stores information about the execution of the current task and its place in the execution graph.  Mainly used
+    * for convenience during lookups.
+    */
+  private final class TrackingInfo(task: Task, parent: Option[GraphNode] = None) {
```

Comment:
Is this parent in the sense of the "enclosing" graph node, or is this just the corresponding graph node for the task?